File: packages/dart/lib/src/word/rand_color.dart
import 'dart:math';

import 'package:randino/src/types.dart';
import 'package:randino/src/word/rand_word.dart';

/// Colours, from the plain ones to the ones with a history.
///
/// [randWord] with the theme already chosen; every other parameter is the same.
/// For the detail form, pass [WordTheme.color] to `randWordDetails` — Dart has
/// no overloads, and twenty-four more functions for it would be twenty-four too many.
///
/// ```dart
/// randColor(language: WordLanguage.ko, count: 3); // [주홍, 연두, 쪽빛]
/// randColor(language: WordLanguage.en, count: 3); // [Crimson, Teal, Ochre]
/// ```
List<String> randColor({
  WordLanguage? language,
  int count = 1,
  RandRealism realism = RandRealism.real,
  RandVocabulary vocabulary = RandVocabulary.full,
  int? minLength,
  int? maxLength,
  String? startsWith,
  bool unique = false,

  /// Where the randomness comes from: `Random.secure()` for a value nobody may
  /// predict, `Random(42)` for one that has to come out the same every run.
  Random? random,
}) => randWord(
  language: language,
  theme: WordTheme.color,
  count: count,
  realism: realism,
  vocabulary: vocabulary,
  minLength: minLength,
  maxLength: maxLength,
  startsWith: startsWith,
  unique: unique,
  random: random,
);

File: packages/javascript/lib/word/randColor.ts
import type { RandThemedWordOptions, WordDetail } from '../_types/global.js';
import { themedWord } from './wordGenerator.js';

/**
 * Colours, from the plain ones to the ones with a history.
 *
 * `randWord({ theme: 'color' })` with the theme already chosen; every other
 * option is the same.
 *
 * @example
 * randColor({ language: 'ko', count: 3 }); // ['주홍', '연두', '쪽빛']
 * randColor({ language: 'en', count: 3 }); // ['Crimson', 'Teal', 'Ochre']
 */
export function randColor(options?: RandThemedWordOptions & { output?: 'value' }): string[];
/**
 * Colour words with the language and theme each one came from.
 *
 * @example
 * randColor({ language: 'ko', output: 'detail' });
 * // [{ word: '주홍', language: 'ko', theme: 'color' }]
 */
export function randColor(options: RandThemedWordOptions & { output: 'detail' }): WordDetail[];
export function randColor(options: RandThemedWordOptions = {}): string[] | WordDetail[] {
	return themedWord('color', options);
}
